src/test/java/hudson/plugins/ec2/AmazonEC2CloudTest.java

```diff
@@ -58,4 +64,18 @@ public void testConfigRoundtrip() throws Exception {
         Cloud actual = r.jenkins.clouds.iterator().next();
         r.assertEqualBeans(orig, actual, "cloudName,region,useInstanceProfileForCredentials,accessId,privateKey,instanceCap,roleArn,roleSessionName");
     }
+
+    /**
+     * Unit tests related to {@link AmazonEC2Cloud}, but do not require a Jenkins instance.
```

Comment:
You might be able to use the `@WithoutJenkins` annotation on the test method without having to make an inner class.

Reply:
when i move the method to the outer class and add `@Test` and `@WithoutJenkins` i end up with this error:

```
[ERROR] Errors: 
[ERROR] hudson.plugins.ec2.AmazonEC2CloudTest.testEC2EndpointURLCreation(hudson.plugins.ec2.AmazonEC2CloudTest)
[ERROR]   Run 1: AmazonEC2CloudTest.testConfigRoundtrip:61 » NullPointer
[ERROR]   Run 2: AmazonEC2CloudTest.testConfigRoundtrip:61 » NullPointer
[ERROR]   Run 3: AmazonEC2CloudTest.testConfigRoundtrip:61 » NullPointer
[ERROR]   Run 4: AmazonEC2CloudTest.testConfigRoundtrip:61 » NullPointer
[ERROR]   Run 5: AmazonEC2CloudTest.testConfigRoundtrip:61 » NullPointer
```

maybe im doing something wrong there?

Reply:
here's the stacktrace:

```
-------------------------------------------------------------------------------
Test set: hudson.plugins.ec2.AmazonEC2CloudTest
-------------------------------------------------------------------------------
Tests run: 5, Failures: 0, Errors: 5, Skipped: 0, Time elapsed: 0.348 s <<< FAILURE! - in hudson.plugins.ec2.AmazonEC2CloudTest
testEC2EndpointURLCreation(hudson.plugins.ec2.AmazonEC2CloudTest)  Time elapsed: 0.342 s  <<< ERROR!
java.lang.NullPointerException
	at jenkins.security.ConfidentialStore.get(ConfidentialStore.java:67)
	at jenkins.security.ConfidentialKey.load(ConfidentialKey.java:47)
	at jenkins.security.CryptoConfidentialKey.getKey(CryptoConfidentialKey.java:41)
	at jenkins.security.CryptoConfidentialKey.decrypt(CryptoConfidentialKey.java:134)
	at hudson.util.HistoricalSecrets.decrypt(HistoricalSecrets.java:49)
	at hudson.util.Secret.decrypt(Secret.java:207)
	at hudson.util.Secret.fromString(Secret.java:239)
	at hudson.plugins.ec2.EC2PrivateKey.<init>(EC2PrivateKey.java:50)
	at hudson.plugins.ec2.EC2Cloud.<init>(EC2Cloud.java:177)
	at hudson.plugins.ec2.AmazonEC2Cloud.<init>(AmazonEC2Cloud.java:74)
	at hudson.plugins.ec2.AmazonEC2CloudTest.testConfigRoundtrip(AmazonEC2CloudTest.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
```